client/gutenberg/extensions/subscriptions/edit.jsx

```diff
@@ -0,0 +1,166 @@
+
+const { getComputedStyle } = window;
+
+const applyFallbackStyles = withFallbackStyles( ( node, ownProps ) => {
+	const { textColor, backgroundColor } = ownProps;
+	const backgroundColorValue = backgroundColor && backgroundColor.color;
+	console.log( backgroundColorValue );
+	const textColorValue = textColor && textColor.color;
+	console.log( textColorValue );
+	// //avoid the use of querySelector if textColor color is known and verify if node is available.
+	const textNode =
+		! textColorValue && node ? node.querySelector( '[contenteditable="true"]' ) : null;
+	// return {
+	// 	fallbackBackgroundColor: backgroundColorValue || ! node ? undefined : getComputedStyle( node ).backgroundColor,
+	// 	fallbackTextColor: textColorValue || ! textNode ? undefined : getComputedStyle( textNode ).color,
+	// };
+} );
+
+console.log( applyFallbackStyles.textColor );
+
+class SubscriptionEdit extends Component {
+	render() {
+		const {
+			attributes,
+			className,
+			isSelected,
+			setAttributes,
+			backgroundColor,
+			textColor,
+			setBackgroundColor,
+			setTextColor,
+			fallbackBackgroundColor,
+			fallbackTextColor,
+		} = this.props;
+		const {
+			subscribe_placeholder,
+			show_subscribers_total,
+			subscriber_count_string,
+			button_background_color,
+			button_text_color,
+			button_large,
+		} = attributes;
+
+		const buttonStyle = {
+			'background-color': button_background_color,
+			color: button_text_color,
+			border: 'none',
+		};
+		// Get the subscriber count so it is available right away if the user toggles the setting
+		this.get_subscriber_count();
+
+		if ( isSelected ) {
+			return (
+				<Fragment>
+					<div className={ className } role="form">
+						{ isSelected && <p role="heading">{ subscriber_count_string }</p> }
+						<Disabled>
+							<TextControl placeholder={ subscribe_placeholder } required onChange={ () => {} } />
+						</Disabled>
+						<Button isDefault isLarge={ button_large } onClick={ null } style={ buttonStyle }>
+							{ __( 'Subscribe' ) }
+						</Button>
+					</div>
+					<InspectorControls>
+						<ToggleControl // Move this back to the block
+							label={ __( 'Show total subscribers' ) }
+							checked={ show_subscribers_total }
+							onChange={ () => {
+								setAttributes( { show_subscribers_total: ! show_subscribers_total } );
+							} }
+						/>
+						<ToggleControl
+							label={ __( 'Use a large button' ) }
+							checked={ button_large }
+							onChange={ () => {
+								setAttributes( { button_large: ! button_large } );
+							} }
+						/>
+						<PanelColorSettings
+							title={ __( 'Button Color Settings' ) }
+							colorSettings={ [
+								{
+									value: backgroundColor.color,
+									label: __( 'Background Color' ),
+									onChange: setBackgroundColor,
+								},
+								{
+									value: textColor.color,
+									label: __( 'Text Color' ),
+									onChange: setTextColor,
+								},
+							] }
+						/>
+						<ContrastChecker
+							{ ...{
```

Comment:
Do we need the `{ ...{ ... }  }` - we could simply pass these as props to the component using the attribute syntax